melusine/models/models_v2/cnn_model.py

```diff
@@ -0,0 +1,267 @@
+import os
+import logging
+import numpy as np
+import pandas as pd
+from typing import List, Dict, Callable
+from tensorflow.keras.layers import Input
+from tensorflow.keras.models import Model
+from tensorflow.keras.preprocessing.sequence import pad_sequences
+
+from melusine.models.models_v2.base_model import BaseMelusineModel
+from melusine.nlp_tools.tokenizer import WordLevelTokenizer
+from melusine.models.models_v2.default_nn_architectures import (
+    default_cnn_archi,
+    default_meta_archi,
+    default_dense_archi,
+    default_multiclass_archi,
+)
+
+logger = logging.getLogger(__name__)
+
+
```

Comment:
all the child classes that inherit `BaseMelusineModel` shall replace `NeuralModel` right ?
And you think about a  `CnnMelusineModel`,  `RnnMelusineModel`, ...
Good to separate BertModels and CNN in my opinion because it will allow you to delete all the `if self.architecture_function.__name__ != "bert_model"` (it occurs 5 times in NeuralModel, I don't like it).

But I think CnnModel and RnnModel should not be separated so far because they have the same methods (the same tokenizer, same metadata, same transform methods, ...)


Or have another step of inheritance : 
`class RNNorCNNMelusineModel(BaseMelusineModel)`
(help me find a better name please ^^)
with the common transform method, the common tokenizer, ...
and two child classes : 
`RnnMelusineModel(RNNorCNNMelusineModel)`
and 
`CnnMelusineModel(RNNorCNNMelusineModel)`

Reply:
Good point, thanks
Actually I think you can already define an RNN model using the Current CnnMelusineModel
by just changing the cnn_archi attribute.

We could try the following:
- Change the class name to be generic (WordEmbeddingBasedMelusineModel? EmbeddingBasedModel? LegacyNlpModel? HunggingfaceIsWayTooSimpleMelusineModel? )
- Add an RNN archi in the default architectures
- Describe how to use an RNN in a tutorial